packages/oats-axios-adapter/src/utils.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Use `URLSearchParams` instead of passing raw data to axios.
+ * Otherwise axios will suffix array query parameters with "[]".
+ * @see https://github.com/axios/axios/issues/2840
+ */
+export function urlSearchParamsSerializer(params: unknown) {
+  if (typeof params !== 'object') {
+    return String(params);
+  }
+  if (params === null) {
+    return '';
+  }
+  if (params instanceof URLSearchParams) {
+    return params.toString();
+  }
+  const queryParams = new URLSearchParams();
+  Object.keys(params).forEach(key => {
+    const value = (params as Record<string, unknown>)[key];
+    for (const item of ([] as unknown[]).concat(value)) {
+      queryParams.append(key, String(item));
```

Comment:
Hey can you explain how this works? Like if the object is {key1: [v1, v2, v3]}, then it seems this would resolve as `key1=v1&key1=v2&key1=v3`